### ISSUE TITLE
Mobile FTUE: Create an Account (meta)

### ISSUE THREAD

**Issue by daniellekirkwood** [vector-im/element-meta#128]
- [x] https://github.com/vector-im/element-android/issues/5277
- [x] https://github.com/vector-im/element-ios/issues/5648

We want to provide the user with a screen to create an account.

Depending on the server chosen there are different options available to users at the sign up stage. Due to this complication users **must** select the server they would like to create their account on before moving forwards with the account creation process. This question confuses new users who are not aware of what this means and can lead to serious detractors and big negative impact on our EPU score (we can see this in the difference between iOS and Android scores today where android users are offered this choice first). To minimise the impact of this decision, reduce the stress it creates, and ensure server options are front and centre for all users, this is our chosen design: 

| Design | 
|----|
| <img width="174" alt="Screenshot 2022-02-16 at 15 09 14" src="https://user-images.githubusercontent.com/89144281/154294682-13b675db-9c84-4d99-8136-7e3bd37ac0fa.png"> |

If the user chooses to edit the server they are taken to the select server screen #126 
When the server is chosen the SSO options may change.

**Comment by daniellekirkwood** [comment]
https://github.com/vector-im/element-android/issues/5277	
https://github.com/vector-im/element-ios/issues/5648

**Comment by daniellekirkwood** [comment]
# Username info

### Spec references: 
- https://spec.matrix.org/latest/#users
- https://spec.matrix.org/v1.2/appendices/#user-identifiers

### TL;DR
- User accounts have unique identifiers that are user IDs (or Matrix User IDs/ MXID).
- This ID is name-spaced to the homeserver like: @username:homeserver.org.
- The length of a userID including "@" must not exceed 255 characters.
- The username section must not be empty and must contain only the characters:  
  - a-z, 0-9, .,  _,  =,  -, and  /

**Comment by daniellekirkwood** [comment]
We should ask the keyboard **not** to capitalise the first letter of the username as capital letters are not allowed in usernames and it has been seen in usertesting to trip people up: 
https://github.com/vector-im/element-ios/issues/5664
https://github.com/vector-im/element-android/issues/5099

**Comment by HarHarLinks** [comment]
![image](https://user-images.githubusercontent.com/2803622/162996426-5584acaf-7702-46fd-9a14-8f2d8cbda92f.png)

I would like to criticize this part:
- while defaulting to `matrix.org` is fine, I dislike element promoting it as if it were the best choice
- wording of "edit" seems questionable, would "change" inflate the button size to much?

Do we only see #126 by clicking edit here? I would argue that #126 should be shown before landing here to inform about (and perhaps encourage) server choice. Is there an overview flow graph somewhere?

**Comment by ouchadam** [comment]
for some extra context, here's the current web account creation (we've aligned quite closely~)

![2022-04-12T16:28:35,847557938+01:00](https://user-images.githubusercontent.com/1848238/162998580-fa477f97-ca07-4317-8963-61881c658717.png)

> Do we only see https://github.com/vector-im/element-meta/issues/126 by clicking edit here? I would argue that https://github.com/vector-im/element-meta/issues/126 should be shown before landing here to inform about (and perhaps encourage) server choice. Is there an overview flow graph somewhere?

correct, the server selection screen is currently only accessible by tapping `EDIT` within the create account screen

| GIF | 
| --- | 
![](https://user-images.githubusercontent.com/1848238/161587211-b032ddb3-0e3f-41da-b8c3-f07e21e6cdaf.gif)

**Comment by HarHarLinks** [comment]
Thank you, Adam. I agree it is well aligned with web.
This may not be the right place for this discussion, and I've criticized this on web before as well, but my points about the extra promotional label to matrix.org and defaulting to the server selection screen (which then lets you select matrix.org if you have nothing else), stand (for all elements).
